src/combinatorics/simple_graph/regularity/uniform.lean

```diff
@@ -80,11 +85,90 @@ begin
   exact zero_lt_one,
 end
 
+variables {G}
+
+lemma exists_nonuniform_witnesses (h : ¬G.is_uniform ε s t) :
+  ∃ s', s' ⊆ s ∧ ∃ t', t' ⊆ t ∧ ↑s.card * ε ≤ s'.card ∧ ↑t.card * ε ≤ t'.card ∧
+    ε ≤ |G.edge_density s' t' - G.edge_density s t| :=
+by { rw is_uniform at h, push_neg at h, exact h }
+
+open_locale classical
+variables (G)
+
+/-- A pair of subsets witnessing the non-uniformity of `(s, t)`. If `(s, t)` is uniform, returns
```

Comment:
Maybe make this documentation slightly clearer that this is some arbitrary pair? The casual reader (like me) might miss that these aren't sets of possible witnesses.